pkg/streamingpromql/engine_test.go

```diff
@@ -861,6 +860,236 @@ func TestRangeVectorSelectors(t *testing.T) {
 	}
 }
 
+func TestSubqueries(t *testing.T) {
+	// This test is based on Prometheus' TestSubquerySelector.
+	data := `load 10s
+	           metric 1 2
+	           http_requests{job="api-server", instance="0", group="production"}	0+10x1000 100+30x1000
+	           http_requests{job="api-server", instance="1", group="production"}	0+20x1000 200+30x1000
+	           http_requests{job="api-server", instance="0", group="canary"}		0+30x1000 300+80x1000
+	           http_requests{job="api-server", instance="1", group="canary"}		0+40x2000`
+
+	opts := NewTestEngineOpts()
+	engine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), log.NewNopLogger())
+	require.NoError(t, err)
+	storage := promqltest.LoadedStorage(t, data)
+	t.Cleanup(func() { storage.Close() })
+
+	testCases := []struct {
+		Query  string
+		Result promql.Result
```

Comment:
(nit) Is it worth carrying all of these `Result`'s? Why not just compare to Prometheus' output? (especially if these exist upstream for correctness)

Reply:
Also we can more easily add NH series to check on those

Reply:
I have found it useful to have the results explicit in this test, as it's been helpful to be able to see the expected results when looking at the test case to better understand what the test is exercising.

Reply:
Fair enough. This tests the results are what we expect. Since it's low hanging fruit, we should also test that Prometheus has the results we expect (ie we should add a comparison anyway).

Reply:
I've added some native histogram tests in ac84802.